src/main/scala/no/ndla/articleapi/service/ReadService.scala

```diff
@@ -8,16 +8,40 @@
 
 package no.ndla.articleapi.service
 
-import no.ndla.articleapi.model.api.Article
+import no.ndla.articleapi.model.{api, domain}
 import no.ndla.articleapi.repository.ArticleRepository
+import no.ndla.articleapi.ArticleApiProperties.{externalApiUrls, resourceHtmlEmbedTag}
+import no.ndla.articleapi.integration.ConverterModule
+
+import scala.collection.JavaConversions._
 
 trait ReadService {
   this: ArticleRepository with ConverterService =>
   val readService: ReadService
 
   class ReadService {
-    def withId(id: Long): Option[Article] =
-      articleRepository.withId(id).map(converterService.toApiArticle)
+    def withId(id: Long): Option[api.Article] =
+      articleRepository.withId(id)
+        .map(addUrlsOnEmbedResources)
+        .map(converterService.toApiArticle)
+
+    private[service] def addUrlsOnEmbedResources(article: domain.Article): domain.Article = {
+      val articleWithUrls = article.content.map(addUrlOnResource)
+      article.copy(content = articleWithUrls)
+    }
+
+    private[service] def addUrlOnResource(content: domain.ArticleContent): domain.ArticleContent = {
+      val doc = ConverterModule.stringToJsoupDocument(content.content)
+      val resourceIdAttrName = "data-resource_id"
+
+      for (el <- doc.select(s"""$resourceHtmlEmbedTag[$resourceIdAttrName]""")) {
+        val (resourceType, id) = (el.attr("data-resource"), el.attr(resourceIdAttrName))
+        el.removeAttr(resourceIdAttrName)
+        el.attr("data-url", s"${externalApiUrls(resourceType)}/$id")
```

Comment:
Det er sant. Dersom `externalApiUrls` ikke inneholder innslag for en gitt ressurstype er det en bug; da kan jeg enten bare droppe å legge på et `data-url` felt, eller kaste en exception. Ulempen med å kaste en exception er at det da ikke vil være mulig å lese artikkelen. Fordelen er selvfølgelig at det er større sannsynlighet for å oppdage feilen.